credentials/credentials_test.go

```diff
@@ -58,4 +62,151 @@ func TestTLSClone(t *testing.T) {
 	if c.Info().ServerName != expectedServerName {
 		t.Fatalf("Change in clone should not affect the original, c.Info().ServerName = %v, want %v", c.Info().ServerName, expectedServerName)
 	}
+
+}
+
+const tlsDir = "../test/testdata/"
+
+type serverHandshake func(net.Conn, *tls.ConnectionState) error
+
+func TestClientHandshakeReturnsAuthInfo(t *testing.T) {
+	var serverConnState tls.ConnectionState
+	done := make(chan error, 1)
```

Comment:
make this a channel of authInfo. The way launchServer does (&serverConnState) is more like C/C++ way instead of Go idiom. In this way, serverHandshake does not need the second param.